src/App.vue

```diff
@@ -99,42 +58,50 @@ export default defineComponent({
       </select>
     </div>
 
-    <div class="error-message" v-if="error">{{ error.message }}</div>
-
-    <GameControl v-else-if="selected_album !== null" :selected_album="selected_album" :refreshSettings="refreshSettings"
-      :addGameHistory="addGameHistory" :hard_mode="hard_mode" />
+    <GameControl @addGameHistory="(game) => { addGameHistory(game), toggleShowStats() }" />
+
+    <div class="error-message" v-if="error">
+      <div v-if="dev_mode">
+        {{ error.message }}
+      </div>
+      <div v-else>
+        Error loading occured
+      </div>
+      <button @click="refreshSettings()">
+        Try Again
+      </button>
+    </div>
 
   </div>
   <div v-if="dev_mode" class="dev-answers">
     <p v-if="selected_album">Album - {{ selected_album.name }}</p>
     <p v-if="selected_album">Artist - {{ selected_album.artist }}</p>
   </div>
 
-
   <div v-if="show_how">
-      <Modal @closemodal="toggleShowHow">
-        <HowtoPlayModal />
-      </Modal>
-    </div>
+    <Modal @closemodal="toggleShowHow">
+      <HowtoPlayModal />
+    </Modal>
+  </div>
 
-    <div v-if="show_stats">
-      <Modal @closemodal="toggleShowStats">
-        <StatsModal :gameHistory="game_history" />
-      </Modal>
-    </div>
+  <div v-if="show_stats">
+    <Modal @closemodal="toggleShowStats">
+      <StatsModal />
+    </Modal>
+  </div>
 
-    <div v-if="show_settings">
-      <Modal @closemodal="toggleShowSettings">
-        <SettingsModal>
-            <template #devSwitch>
-              <ToggleSwitch :value="dev_mode" @updateValue="toggleDevMode"/> 
-            </template>
-            <template #modeSwitch>
-              <ToggleSwitch :value="hard_mode" @updateValue="toggleHardMode"/>
-            </template>
-        </SettingsModal>
-      </Modal>
-    </div>
+  <div v-if="show_settings">
+    <Modal @closemodal="toggleShowSettings">
+      <SettingsModal>
+        <template #devSwitch>
+          <ToggleSwitch :value="dev_mode" @updateValue="toggleDevMode" />
+        </template>
+        <template #modeSwitch>
+          <ToggleSwitch :value="hard_mode" @updateValue="toggleHardMode" />
+        </template>
+      </SettingsModal>
```

Comment:
Now that `dev_mode` and `hard_mode` are accessible in SettingsModal directly via `useSettings` I would switch to that from this slots pattern